### ISSUE TITLE
Context menu save image crashing off-main thread

### ISSUE THREAD

**Issue by garvankeeley** [mozilla-mobile/firefox-ios#6482]
This should crash in the context menu for save image if the user denies the request to access the photo library.

Key info
```
#34	(null) in closure #2 in closure #9 in closure #2 in BrowserViewController.webView(_:contextMenuConfiguratio... ()
#35	(null) in partial apply for closure #3 in closure #9 in closure #2 in BrowserViewController.webView(_:conte... ()
#36	(null) in thunk for @escaping @callee_guaranteed (@unowned PHAuthorizationStatus) -> () ()
```

Full stacktrace:
```
ast Exception Backtrace:
0   CoreFoundation                	0x1b38cb5f0 __exceptionPreprocess + 224 (NSException.m:199)
1   libobjc.A.dylib               	0x1b35edbcc objc_exception_throw + 56 (objc-exception.mm:565)
2   Foundation                    	0x1b3da2dac _AssertAutolayoutOnAllowedThreadsOnly + 420 (NSISEngine.m:0)
3   Foundation                    	0x1b3bafc2c -[NSISEngine _optimizeWithoutRebuilding] + 68 (NSISEngine.m:1640)
4   Foundation                    	0x1b3bafb48 -[NSISEngine optimize] + 112 (NSISEngine.m:1614)
5   Foundation                    	0x1b3baf7bc -[NSISEngine performPendingChangeNotifications] + 112 (NSISEngine.m:632)
6   UIKitCore                     	0x1b7e10e1c -[UIView(Hierarchy) layoutBelowIfNeeded] + 240 (UIView.m:11982)
7   UIKitCore                     	0x1b7759cf0 -[UIInputSetHostView layoutIfNeeded] + 96 (UIInputWindowController.m:451)
8   UIKitCore                     	0x1b77609f0 __82-[UIInputWindowController updateToPlacement:withNormalAnimationsAndNotifications:]_block_invoke + 84 (UIInputWindowController.m:1987)
9   UIKitCore                     	0x1b7e175ec +[UIView(Animation) performWithoutAnimation:] + 96 (UIView.m:13773)
10  UIKitCore                     	0x1b775b050 -[UIInputWindowController performWithoutAppearanceCallbacks:] + 48 (UIInputWindowController.m:877)
11  UIKitCore                     	0x1b775b098 -[UIInputWindowController performWithoutCallbacksOrNotifications:] + 48 (UIInputWindowController.m:884)
12  UIKitCore                     	0x1b776098c -[UIInputWindowController updateToPlacement:withNormalAnimationsAndNotifications:] + 196 (UIInputWindowController.m:1993)
13  UIKitCore                     	0x1b7536480 -[UIInputViewAnimationControllerBasic prepareAnimationWithHost:startPlacement:endPlacement:] + 40 (UIInputViewAnimationStyle.m:179)
14  UIKitCore                     	0x1b775f904 __77-[UIInputWindowController moveFromPlacement:toPlacement:starting:completion:]_block_invoke.887 + 52 (UIInputWindowController.m:1825)
15  UIKitCore                     	0x1b77e6c68 -[_UIRemoteKeyboards ignoreLayoutNotifications:] + 40 (_UIRemoteKeyboards.m:1616)
16  UIKitCore                     	0x1b775f508 -[UIInputWindowController moveFromPlacement:toPlacement:starting:completion:] + 1480 (UIInputWindowController.m:1824)
17  UIKitCore                     	0x1b7764878 __43-[UIInputWindowController setInputViewSet:]_block_invoke_7 + 96 (UIInputWindowController.m:2746)
18  UIKitCore                     	0x1b775b0e8 -[UIInputWindowController performOperations:withTemplateNotificationInfo:] + 56 (UIInputWindowController.m:891)
19  UIKitCore                     	0x1b7764580 -[UIInputWindowController setInputViewSet:] + 1264 (UIInputWindowController.m:2745)
20  UIKitCore                     	0x1b775ebac -[UIInputWindowController performOperations:withAnimationStyle:] + 52 (UIInputWindowController.m:1685)
21  UIKitCore                     	0x1b7475d08 -[UIInputResponderController setKeyWindowSceneInputViews:animationStyle:] + 2160 (UIInputResponderController.m:1083)
22  UIKitCore                     	0x1b7475474 -[UIInputResponderController setInputViews:animationStyle:] + 212 (UIInputResponderController.m:0)
23  UIKitCore                     	0x1b747654c -[UIInputResponderController setInputViews:animated:] + 96 (UIInputResponderController.m:1134)
24  UIKitCore                     	0x1b74765b4 -[UIInputResponderController setInputViews:] + 76 (UIInputResponderController.m:1139)
25  UIKitCore                     	0x1b747411c -[UIInputResponderController _reloadInputViewsForKeyWindowSceneResponder:] + 3120 (UIInputResponderController.m:765)
26  UIKitCore                     	0x1b74734d0 -[UIInputResponderController _reloadInputViewsForResponder:] + 160 (UIInputResponderController.m:0)
27  UIKitCore                     	0x1b7478a14 -[UIInputResponderController _preserveInputViewsWithId:animated:reset:] + 128 (UIInputResponderController.m:1607)
28  UIKitCore                     	0x1b7344088 -[UIViewController _presentViewController:modalSourceViewController:presentationController:animat... + 1212 (UIViewController.m:7020)
29  UIKitCore                     	0x1b7345b08 -[UIViewController _presentViewController:withAnimationController:completion:] + 4228 (UIViewController.m:7378)
30  UIKitCore                     	0x1b7348014 __63-[UIViewController _presentViewController:animated:completion:]_block_invoke + 104 (UIViewController.m:7736)
31  UIKitCore                     	0x1b734851c -[UIViewController _performCoordinatedPresentOrDismiss:animated:] + 508 (UIViewController.m:7837)
32  UIKitCore                     	0x1b7347f64 -[UIViewController _presentViewController:animated:completion:] + 196 (UIViewController.m:7743)
33  UIKitCore                     	0x1b73481d0 -[UIViewController presentViewController:animated:completion:] + 164 (UIViewController.m:7781)
34  Client                        	0x100251100 closure #2 in closure #9 in closure #2 in BrowserViewController.webView(_:contextMenuConfiguratio... + 516
35  Client                        	0x10025f308 partial apply for closure #3 in closure #9 in closure #2 in BrowserViewController.webView(_:conte... + 44
36  Client                        	0x1003b53f0 thunk for @escaping @callee_guaranteed (@unowned PHAuthorizationStatus) -> () + 40
37  Photos                        	0x1bef5c110 __39+[PHPhotoLibrary requestAuthorization:]_block_invoke + 64 (PHPhotoLibrary.m:282)
38  AssetsLibraryServices         	0x1c8273274 __79-[PLPrivacy _isPhotosAccessAllowedWithScope:forceHandler:accessAllowedHandler:]_block_invoke.14 + 516 (PLPrivacy.m:144)
39  AssetsLibraryServices         	0x1c823e0d4 __pl_dispatch_async_block_invoke + 36 (PLDispatchHelpers.m:23)
40  libdispatch.dylib             	0x1b35909a8 _dispatch_call_block_and_release + 24 (init.c:1408)
41  libdispatch.dylib             	0x1b3591524 _dispatch_client_callout + 16 (object.m:495)
42  libdispatch.dylib             	0x1b353d8a4 _dispatch_lane_serial_drain$VARIANT$mp + 608 (inline_internal.h:2484)
43  libdispatch.dylib             	0x1b353e294 _dispatch_lane_invoke$VARIANT$mp + 416 (queue.c:3863)
44  libdispatch.dylib             	0x1b354778c _dispatch_workloop_worker_thread + 588 (queue.c:6445)
45  libsystem_pthread.dylib       	0x1b35e2b74 _pthread_wqthread + 272 (pthread.c:2351)
46  libsystem_pthread.dylib       	0x1b35e5740 start_wqthread + 8
```

**Comment by garvankeeley** [comment]
#6556 fixed this

**Comment by SimonBasca** [comment]
Verified fix on 26.0(17952).
